### PR TITLE
Revocation memory

### PR DESCRIPTION
**Pull request by marnovandermaas**
This commit creates a two-ported SRAM to be used as revocation tags.

I've tested that regular CHERI code still works but not sure whether the revocation tags themselves work.

### CODE REVIEW

**Comment by marnovandermaas** [comment]
In my latest force push I added a basic SRAM test for the revocation memory.

**Review by nbdd0121 — APPROVED** [review]
The RTL changes look alright. I haven't checked or tested the SW. There seems quite a bit SW duplication but that could be left to a later cleanup PR.

File: sw/cheri/revocation_test/boot.S [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    .type start,@function
+start:
+	// ca0 (first argument) contains the read-write root
+	cspecialr        ca0, mtdc
+
+	// Zero the tag memory
+	//li               a1, 0x200fe000
+	//csetaddr         ca0, ca0, a1
+	//li               a1, 0x20100000
+	//cjal             zero_memory
```
Comment (HU90m) [inline sw/cheri/revocation_test/boot.S:16]: Could these commented out sections be removed?

**Review by HU90m — APPROVED** [review]

File: sw/cheri/revocation_test/boot.S (hunk 1 above)
Comment (marnovandermaas) [reply sw/cheri/revocation_test/boot.S:16]: I think so, this is copied from the CHERI sanity test and I left it in for reference. Let's do a more general cleanup in the software in a later PR.

**Pull request merged**